core/src/main/java/org/fao/geonet/api/records/attachments/StoreUtils.java

```diff
@@ -68,12 +68,9 @@ public static void copyDataDir(ServiceContext context, int oldMetadataId, int ne
     public static void copyDataDir(ServiceContext context, String oldUuid, String newUuid, boolean newApproved) throws Exception {
         final Store store = context.getBean("resourceStore", Store.class);
         for (MetadataResourceVisibility visibility: MetadataResourceVisibility.values()) {
-            final List<MetadataResource> resources = store.getResources(context, oldUuid, visibility, null, true);
-            for (MetadataResource resource: resources) {
-                try (Store.ResourceHolder holder = store.getResource(context, oldUuid, visibility, resource.getFilename(), true)) {
-                    store.putResource(context, newUuid, holder.getPath(), visibility, newApproved);
-                }
-            }
+            //Copy from approved copy to working copy
+            boolean oldApproved = true;
+            store.copyResources(context, oldUuid, newUuid, visibility, oldApproved, newApproved);
```

Comment:
Probably I'm missing something as haven't check in detail, but not sure if this is going to work with the `FilesystemStore`? In previous code calls ` store.putResource` that managed to copy the resources:

https://github.com/geonetwork/core-geonetwork/blob/17aa5333aa16546ca004e96d8429eaf9db95f931/core/src/main/java/org/fao/geonet/api/records/attachments/FilesystemStore.java#L187-L196

But in this new version calls `store.copyResources` that seem empty in `FilesystemStore`.

https://github.com/geonetwork/core-geonetwork/pull/5776/files#diff-27baa8be68aa76d7a5f25db314795edb869f980119dc5d781e82aea9d0f5611eR187-R191

Reply:
I agree, I believe this logic will break all other file stores.

It looks like the old logic should have been moved to the new "public void copyResources" within AbstractStore so that all other file stores would use that default logic. And all the code changes with "// TODO Auto-generated method stub" should be removed so that they inherit the default AbstractStore implementation.

Reply:
I will try to move the old code to general block to all other store types.

Reply:
Move StoreUtils.copyDataDir() implementations to AbstractStore.copyResources as default implementation for all other store type other than CmisStore. I tested locally for CMIS and FilesystemStore